### PR TITLE
Feature/celo cip 64 support

### PR DESCRIPTION
**Pull request by yerdua**

### CODE REVIEW

File: src/chains/celo/parsers.ts [hunk 1]
```diff
@@ -121,3 +131,74 @@ function parseTransactionCIP42(
...
+    maxFeePerGas,
+    gas,
+    feeCurrency,
+    to,
+    value,
+    data,
+    accessList,
+    v,
+    r,
+    s,
```
Comment (aaronmgdr) [inline src/chains/celo/parsers.ts]: the order here i think should have feeCurrency after access list

File: src/chains/celo/serializers.ts [hunk 2]
```diff
@@ -96,22 +109,54 @@ function serializeTransactionCIP42(
...
+    toHex(chainId),
+    nonce ? toHex(nonce) : '0x',
+    maxPriorityFeePerGas ? toHex(maxPriorityFeePerGas) : '0x',
+    maxFeePerGas ? toHex(maxFeePerGas) : '0x',
+    gas ? toHex(gas) : '0x',
+    feeCurrency ?? '0x',
+    to ?? '0x',
+    value ? toHex(value) : '0x',
+    data ?? '0x',
+    serializeAccessList(accessList),
```
Comment (aaronmgdr) [inline src/chains/celo/serializers.ts:133]: https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0064.md

the cip specifies the order as fee currency coming after accessList.

**Comment by aaronmgdr** [comment]
apart from the ordering comments looking good

File: src/chains/celo/serializers.ts [hunk 3]
```diff
@@ -18,20 +18,32 @@ import {
...
-      signature,
-    )
-
-  // Handle other transaction types
+  if ('maxFeePerGas' in tx && 'maxPriorityFeePerGas' in tx) {
+    // process as CIP42 if any of these fields are present. realistically gatewayfee is not used but is part of spec
+    if (
+      'gatewayFee' in tx ||
+      'gatewayFeeRecipient' in tx ||
+      ('feeCurrency' in tx && tx.type && tx.type === 'cip42')
```
Comment (nicolasbrugneaux) [inline src/chains/celo/serializers.ts]: I wouldn't remove `isCIP42`, I'd have a `isCIP64` method to hide implementation details here
```ts

if (isCIP4(tx)) {
  return serializeTransactionCIP64(
    tx as TransactionSerializableCIP64,
    signature,
  )
}
if (isCIP42(tx)) {
  ...
}
```

File: src/chains/celo/parsers.ts (hunk 1 above)
Comment (yerdua) [reply src/chains/celo/parsers.ts]: oops. thanks!

File: src/chains/celo/formatters.test.ts [hunk 4]
```diff
@@ -237,30 +237,30 @@ describe('block', () => {
...
+      {
+        "blockHash": "0xac8c9bc3b84e103dc321bbe83b670e425ff68bfc9a333a4f1b1b204ad11c583d",
+        "blockNumber": 16645775n,
+        "chainId": undefined,
+        "ethCompatible": false,
+        "feeCurrency": null,
+        "from": "0x045d685d23e8aa34dc408a66fb408f20dc84d785",
+        "gas": 1527520n,
+        "gasPrice": 562129081n,
+        "gatewayFee": "0x0",
```
Comment (yerdua) [inline src/chains/celo/formatters.test.ts]: I don't know why or if this change is needed. I'd like confirmation about whether this change was right, or if I broke something elsewhere.

Comment (yerdua) [reply src/chains/celo/formatters.test.ts]: A test is failing at `src/actions/public/getBlock.test.ts:54:27`, in a way that looks the same as this. So, I'm inclined to believe that I somehow broke this.

**Comment by jxom** [comment]
Would love to see this PR'd upstream into Viem!

**Comment by yerdua** [comment]
> Would love to see this PR'd upstream into Viem!

Thanks for the encouragement! PR for upstream is here: https://github.com/wagmi-dev/viem/pull/1379

**Comment by aaronmgdr** [comment]
closing in favor of https://github.com/wagmi-dev/viem/pull/1379

**Pull request closed without merging**